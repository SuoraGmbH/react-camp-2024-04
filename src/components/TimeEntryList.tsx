import React from "react";
import TimeEntry from "../domain/TimeEntry.ts";
import { TimeEntryView } from "./TimeEntryView.tsx";

interface Props {
  timeEntries: readonly TimeEntry[];
}

export const TimeEntryList: React.FunctionComponent<Props> = ({
  timeEntries,
}) => {
  return (
    <div>
      {timeEntries.map((timeEntry) => (
        <TimeEntryView key={timeEntry.id} timeEntry={timeEntry} />
      ))}
    </div>
  );
};
