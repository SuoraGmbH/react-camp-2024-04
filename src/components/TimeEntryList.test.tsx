import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { TimeEntryList } from "./TimeEntryList.tsx";

test("ðŸ§¯", () => {
  render(<TimeEntryList timeEntries={[]} />);
});

it("renders the given items", () => {
  const { container } = render(
    <TimeEntryList
      timeEntries={[
        {
          id: "5049f547-7498-4e2c-a535-29ae338428b9",
          start: new Date("2024-04-17 11:44:00"),
          end: new Date("2024-04-17 11:44:00"),
          comment: "This is just for tests",
        },

        {
          id: "5049f547-7498-4e2c-a535-29ae338428b5",
          start: new Date("2024-04-17 11:44:00"),
          end: new Date("2024-04-17 11:44:00"),
          comment: "This is the second time entry",
        },
      ]}
    />,
  );

  expect(container.textContent).toMatchInlineSnapshot(
    `"This is just for testsThis is the second time entry"`,
  );
});
