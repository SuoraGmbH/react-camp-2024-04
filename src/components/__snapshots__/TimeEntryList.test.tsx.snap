// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the given items 1`] = `
<div>
  <div>
    <div>
      This is just for tests
    </div>
    <div>
      This is the second time entry
    </div>
  </div>
</div>
`;
